### PR TITLE
March roundup

### PR DESCRIPTION
**Pull request by tyhoff**
Didn't get a chance to write descriptions about the bottom three. Probably a good thing because there would be plenty of sass about them.

### CODE REVIEW

File: _drafts/2020-03-31-march-2020-roundup.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+* [Baselibc](https://github.com/PetteriAimonen/Baselibc)
+* [Defer](https://github.com/moon-chilled/Defer) - A one line library which allows statements to run a function at return time, "generally to do cleanup". 
+* [Pigweed by Google](https://pigweed.googlesource.com/pigweed/pigweed/) - A new project from an embedded team within Google.It sounds like much of this code is used across internal teams within Google, but it has been polished, packaged up, and open-sourced! We're excited to see where this goes.
+
+## News
+
+* [The SEGGER Compiler](https://blog.segger.com/the-segger-compiler/)
+* [Introducing Amazon Common Software for Devices (Preview)](https://developer.amazon.com/en-US/blogs/alexa/device-makers/2020/02/introducing-amazon-common-software-for-devices-preview)
+* [An HP Enterprise SSD will brick after 40,000 hours](https://support.hpe.com/hpesc/public/docDisplay?docId=a00097382en_us)
```
Comment (chrisc11) [inline _drafts/2020-03-31-march-2020-roundup.md:54]: amazing

File: _drafts/2020-03-31-march-2020-roundup.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  Fascinating post about how to use custom ELF sections to build a plug-in system, with a description about how this applies to the Linux Kernel.
+* [Static analysis in GCC 10](https://developers.redhat.com/blog/2020/03/26/static-analysis-in-gcc-10/) by David Malcolm  
+  We **love** GCC and Clang, and we are always excited to see them have new features.
+* [async/await on Embedded Rust](https://ferrous-systems.com/blog/async-on-embedded/) by Ferrous Systems  
+  Using async/await on an ARM Cortex-M, complete with an [open-source example](https://github.com/ferrous-systems/async-on-embedded/). Definitely worth the read.
+* [Mastering Embedded Linux Series](https://www.thirtythreeforty.net/posts/2020/03/mastering-embedded-linux-part-4-adding-features/) by George Hilliard
+
+## Neat Open Source Projects
+
+* [Baselibc](https://github.com/PetteriAimonen/Baselibc)
```
Comment (chrisc11) [inline _drafts/2020-03-31-march-2020-roundup.md:46]: Maybe a one-liner here? tangentially related but not super new, I found https://github.com/ARMmbed/minimal-printf/ recently via grep.app, hadn't seen that repo before.  readable printf impl and pretty low memory footprint!

File: _drafts/2020-03-31-march-2020-roundup.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+* [PlatformIO x Renode Demo - Zephyr Build and Debug on SiFive FE310](https://www.youtube.com/watch?v=EdC3kFZxSzI&feature=emb_logo)  
+  This demo has every technical buzzword that people are excited about. PlatformIO, Renode, Zephyr, RISC-V, SiFive...
+* [Exception(al) Failure - Breaking the STM32F1 Read-Out Protection](https://blog.zapb.de/stm32f1-exceptional-failure/) by Marc Schink & Johannes Obermaier.  
+  A wonderfully technical and professional vulnerability write-up. Source-code is also available.
+* [ARM GCC Inline Assembler Cookbook](http://www.ethernut.de/en/documents/arm-inline-asm.html)  
+  I stumbled across this a couple weeks back and figured I'd throw it in the pile.
+* [Hacking Your ELF For Fun And Profit](https://mgalgs.github.io/2013/05/10/hacking-your-ELF-for-fun-and-profit.html) by Mitch Humpherys  
+  Fascinating post about how to use custom ELF sections to build a plug-in system, with a description about how this applies to the Linux Kernel.
+* [Static analysis in GCC 10](https://developers.redhat.com/blog/2020/03/26/static-analysis-in-gcc-10/) by David Malcolm  
+  We **love** GCC and Clang, and we are always excited to see them have new features.
```
Comment (chrisc11) [inline _drafts/2020-03-31-march-2020-roundup.md:39]: <3

File: _drafts/2020-03-31-march-2020-roundup.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+about this past month.
+<!-- excerpt end -->
+
+We hope you enjoy these links, and look forward to hearing what you've been
+reading in the comments.
+
+## Articles & Learning
+
+* [Security Analysis of the Solo Firmware](https://solokeys.com/blogs/news/security-analysis-of-the-solo-firmware-by-doyensec) by Emanuele Cesena.  
+  A nice summary of a security audit of the firmware for a line open source FIDO2 security keys by SoloKeys. The post explains a Downgrade Attack.
```
Comment (franc0is) [inline _drafts/2020-03-31-march-2020-roundup.md:18]:
```suggestion
  A nice summary of a security audit of the firmware for an open source FIDO2 security keys by SoloKeys. The post explains a Downgrade Attack.
```

File: _drafts/2020-03-31-march-2020-roundup.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+* [Making it easier to make Things: WebAssembly and the Internet of Things](https://www.youtube.com/watch?v=oky3FdsTuUM) by Jonathan Beri.  
+  An awesome 30-minute talk about running [Wasm3](https://github.com/wasm3/wasm3) on embedded devices such as an ESP32. It's well worth the watch.
+* [Rewriting m4vgalib in Rust](https://cliffle.com/blog/m4vga-in-rust/) by Cliff Biffle  
+  A fantastic journey through rewriting an embedded graphics demo in Rust from C++ by a well-known programmer. It answers all the important questions like how to not use a standard library, memory consumption, and impact to binary size.
```
Comment (franc0is) [inline _drafts/2020-03-31-march-2020-roundup.md:22]:
```suggestion
  A fantastic journey through rewriting an embedded graphics demo in Rust from C++ by a well-known programmer. It answers all the important questions like how to not use a standard library, memory consumption, and impact on binary size.
```

File: _drafts/2020-03-31-march-2020-roundup.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+* [Unleashing Mayhem over Bluetooth Low Energy](https://asset-group.github.io/disclosures/sweyntooth/)  
+  A deep dive analysis into BLE vulnerabilities into popular BLE chips from Dialog, STMicro, and Cypress chips.
```
Comment (franc0is) [inline _drafts/2020-03-31-march-2020-roundup.md:24]:
```suggestion
  A deep dive analysis into BLE vulnerabilities in popular BLE chipsets from Dialog, STMicro, and Cypress chips. 
```

File: _drafts/2020-03-31-march-2020-roundup.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+* [Introduction to Trusted Execution Environment and ARM's TrustZone](https://embeddedbits.org/introduction-to-trusted-execution-environment-tee-arm-trustzone/) by Sergio Prado
+* [How to Port a Board for RIOT](https://blog.martine-lenders.eu/riot-board-en.html) by Martine Lenders  
+  RIOT being the [RIOT RTOS](https://riot-os.org/). I felt this was post-worthy because one can never read too much about driver abstraction in large projects.
+* [PlatformIO x Renode Demo - Zephyr Build and Debug on SiFive FE310](https://www.youtube.com/watch?v=EdC3kFZxSzI&feature=emb_logo)  
+  This demo has every technical buzzword that people are excited about. PlatformIO, Renode, Zephyr, RISC-V, SiFive...
+* [Exception(al) Failure - Breaking the STM32F1 Read-Out Protection](https://blog.zapb.de/stm32f1-exceptional-failure/) by Marc Schink & Johannes Obermaier.  
+  A wonderfully technical and professional vulnerability write-up. Source-code is also available.
+* [ARM GCC Inline Assembler Cookbook](http://www.ethernut.de/en/documents/arm-inline-asm.html)  
+  I stumbled across this a couple weeks back and figured I'd throw it in the pile.
```
Comment (franc0is) [inline _drafts/2020-03-31-march-2020-roundup.md:35]:
```suggestion
  Though this one isn't new, we stumbled upon it recently. Great reference on using inline assembly.
```

**Pull request merged**